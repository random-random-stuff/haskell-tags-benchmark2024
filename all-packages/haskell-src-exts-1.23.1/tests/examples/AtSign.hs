{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE TypeApplications #-}
{- |
Accelerate interface to the native CUDA implementation
of the Fourier Transform provided by the CUFFT library.
-}
module Data.Array.Accelerate.CUFFT.Private where


transform hndl@(Handle fallback mode width _) =
   wrap mode (A.constant width) $
   A.foreignAcc
      (AF.CUDAForeignAcc "transformForeign" $ transformForeign hndl)
      (unwrap mode (A.constant width) fallback)


