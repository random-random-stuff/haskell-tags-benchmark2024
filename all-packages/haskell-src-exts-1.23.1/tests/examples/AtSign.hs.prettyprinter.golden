{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE TypeApplications #-}
module Data.Array.Accelerate.CUFFT.Private where
transform hndl@(Handle fallback mode width _)
  = wrap mode (A.constant width) $
      A.foreignAcc
        (AF.CUDAForeignAcc "transformForeign" $ transformForeign hndl)
        (unwrap mode (A.constant width) fallback)
